src/string_bytes.cc

```diff
@@ -695,17 +681,14 @@ Local<Value> StringBytes::Encode(Isolate* isolate,
 Local<Value> StringBytes::Encode(Isolate* isolate,
                                  const uint16_t* buf,
                                  size_t buflen) {
+  // Node's "ucs2" encoding expects LE character data inside a
+  // Buffer, so we need to reorder on BE platforms.  See
+  // http://nodejs.org/api/buffer.html regarding Node's "ucs2"
+  // encoding specification
+  if (IsBigEndian())
+    SwapBytes16(const_cast<char*>(reinterpret_cast<const char*>(buf)), buflen);
```

Comment:
No, I just botched and misread the pointer reassignment in the original code. Will fix. Thanks for catching @bnoordhuis.